gtwrap/interface_parser/enum.py

```diff
@@ -0,0 +1,58 @@
+"""
+GTSAM Copyright 2010-2020, Georgia Tech Research Corporation,
+Atlanta, Georgia 30332-0415
+All Rights Reserved
+
+See LICENSE for the license information
+
+Parser class and rules for parsing C++ enums.
+
+Author: Varun Agrawal
+"""
+
+from pyparsing import Optional, alphas, delimitedList, pyparsing_common
+
+from .tokens import ENUM, EQUAL, IDENT, LBRACE, RBRACE, SEMI_COLON
+from .utils import collect_namespaces
+
+
+class Enumerator:
+    """
+    Rule to parse an enumerator inside an enum.
+    """
+    rule = (IDENT("name") +
+            Optional(EQUAL + (pyparsing_common.signed_integer ^ alphas))
+            ).setParseAction(lambda t: Enumerator(t.name))
```

Comment:
should the optional definition value also get saved?

Reply:
It doesn't seem like Pybind11 supports that. Can you confirm please?

Reply:
oh ya i'm dumb - of course it shouldn't since it's compiled into the cpp code.

In which case, it shouldn't really be an option here either right?

Reply:
Yeah but if someone copy-pastes their enum into the interface file, this will allow it to work. It will just throw away the custom values.

Reply:
but won't it be confusing if I think I can change the enum values by coding them in the interface file, then pull my hair out trying to figure out why it doesn't work?  For example,
c++ header file:
```c++
enum Foo {
  a = 1,
  b = 2,
  c = 3
};
```
.i file:
```c++
enum Foo {
  a = 101,
  b = 102,
  c = 103
}
```
python:
```python
state = Foo.a
if state == 101:
  print('good')
else:
  print('what the heck is going on!?!?!?! *cue hours of debugging*')
```

Since there are several other instances where the interface file syntax differs from a header file syntax to clue the user as to the wrapping mechanics (e.g. `boost::shared_ptr`), do you think it would be reasonable to include this as an instance where eliminating the possibility of someone writing invalid code outweighs the inconvenience of a minor modification that has to happen during copy/paste?

Reply:
Is that really what happens? I tried it out and it doesn't seem to map to any particular number, though I can double check and update in that case. From what I understand, it doesn't matter what you set in the interface file, it always maps to the underlying value.

Actually, let's just remove it. Redefining in the interface file is a bad idea in general.